### PR TITLE
Spine v0.7.0

### PR DESCRIPTION
**Pull request by dmdashenkov**
#### Introducing JDBC storage v0.7.0.
 - `EventStreamQuery` complete support (with `FieldFilter`s for the event message and `EventContext`)
 - Code style issues fixes
 - `Sql` class for storing all the SQL keywords and signs used in the project
 - `Guava v20.0`

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol Can we disable string literal duplicate check for this repo, since we have a lot of duplications when building the SQL queries?

**Comment by armiol** [comment]
@dmdashenkov please use an appropriate warning suppression (annotation-based one) in the code instead. Remember though you should add a comment for each of the suppressions.

**Comment by dmdashenkov** [comment]
@armiol PTAL

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/util/ConnectionWrapper.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public PreparedStatement prepareStatement(String sql) {
             final PreparedStatement statement = connection.prepareStatement(sql);
             return statement;
         } catch (SQLException e) {
-            throw new DatabaseException(e);
+            throw new DatabaseException("Executing statement " + sql, e);
```
Comment (armiol) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/util/ConnectionWrapper.java]: I think it's `"Error executing statement " + ...`.

File: rdbms/src/test/java/org/spine3/server/storage/jdbc/SqlShould.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    @Test(expected = IllegalArgumentException.class)
+    public void fail_to_generate_zero_placeholders() {
+        final int count = 0;
+        Sql.nPlaceholders(count);
+    }
+
+    @SuppressWarnings("DuplicateStringLiteralInspection")
+    @Test
+    public void provide_valid_sql_tokens() {
+        final String createTAbleExpected = "CREATE TABLE";
```
Comment (armiol) [inline rdbms/src/test/java/org/spine3/server/storage/jdbc/SqlShould.java]: `createTAbleExpected` has a typo.

File: rdbms/src/test/java/org/spine3/server/storage/jdbc/command/query/Given.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ private Given() {
     }
 
     @SuppressWarnings("StaticVariableUsedBeforeInitialization")
```
Comment (armiol) [inline rdbms/src/test/java/org/spine3/server/storage/jdbc/command/query/Given.java]: Looks like this suppression makes not much sense, as the `loggerMock` is initialized above (e.g. `loggerMock = null`).

File: rdbms/src/test/java/org/spine3/server/storage/jdbc/aggregate/query/Given.java [hunk 4]
```diff
@@ -34,7 +34,7 @@
  * @author Andrey Lavrov
  */
 @SuppressWarnings("UtilityClass")
-/* package */ class Given {
+class Given {
 
     @SuppressWarnings("StaticNonFinalField")
     private static Logger loggerMock = null;
```
Comment (armiol) [inline rdbms/src/test/java/org/spine3/server/storage/jdbc/aggregate/query/Given.java]: Please check whether we actually use this mock in assertions in the tests. If not, let's kill this field..

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+package org.spine3.server.storage.jdbc;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Set of enums and utilities for constructing the SQL sentences.
+ * <p>Contains most valid SQL keywords/key-phrases, operators and punctuation.
+ *
+ * <p>All the {@code enum} values have a valid token string representation, i.e. {@link Enum#toString() toString()}
```
Comment (dmdashenkov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java]: The `Sql` class does not override `toString` method. It's overridden in the `enum`s themselves. Should we add links to the implementations in all the enums?

File: rdbms/src/test/java/org/spine3/server/storage/jdbc/aggregate/query/Given.java (hunk 4 above)
Comment (dmdashenkov) [inline rdbms/src/test/java/org/spine3/server/storage/jdbc/aggregate/query/Given.java]: Yes, we use it in constructions like `verify(loggerMock).error(...)`.

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/event/JdbcEventStorage.java [hunk 6]
```diff
@@ -137,13 +165,142 @@ public void close() throws DatabaseException {
...
+            final Collection<Message> expectedValues = Collections2.transform(expectedAnys, ANY_UNPACKER);
+            Message actualValue;
+            try {
+                final Class<?> messageClass = object.getClass();
+                final Method fieldGetter = messageClass.getDeclaredMethod(fieldGetterName);
+                actualValue = (Message) fieldGetter.invoke(object);
+                if (actualValue instanceof Any) {
+                    actualValue = AnyPacker.unpack((Any) actualValue);
+                }
+            } catch (@SuppressWarnings("OverlyBroadCatchBlock") ReflectiveOperationException e) {
```
Comment (dmdashenkov) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/event/JdbcEventStorage.java]: Done

**Comment by dmdashenkov** [comment]
@armiol PTAL again.

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+        @Override
+        public String toString() {
+            return ' ' + name() + ' ';
+        }
+    }
+
+    /**
+     * Set of punctuation signs used in SQL:
+     * <ul>
```
Comment (armiol) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java]: Please allow an empty line above `<ul>`.

File: rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+     *     <li>Operators: equal, not equal, comparison operators;
+     *     <li>Signs: comma, brackets, semicolon.
```
Comment (armiol) [inline rdbms/src/main/java/org/spine3/server/storage/jdbc/Sql.java]: Should go `Punctuation:`.

**Pull request merged**